lib/matplotlib/pyplot.py

```diff
@@ -230,24 +230,51 @@ def switch_backend(newbackend):
     _backend_mod, new_figure_manager, draw_if_interactive, _show = pylab_setup()
 
 
-def show(*args, **kw):
+def show(block=None):
     """
-    Display a figure.
-    When running in ipython with its pylab mode, display all
-    figures and return to the ipython prompt.
+    Draw and display all figures.
 
-    In non-interactive mode, display all figures and block until
-    the figures have been closed; in interactive mode it has no
-    effect unless figures were created prior to a change from
-    non-interactive to interactive mode (not recommended).  In
-    that case it displays the figures but does not block.
+    Mainly used in non-interactive mode to draw and display
+    the figures. It usually does nothing in interactive mode.
 
-    A single experimental keyword argument, *block*, may be
-    set to True or False to override the blocking behavior
-    described above.
+    Parameters
+    ----------
+
+    block : bool or None
+        True means that the execution is blocked until the figures are closed
+        and False means that the python execution continues after the figures
+        are displayed.
+
+        The default behavior is different in interactive and non-interactive
```

Comment:
Shorter:

The default is *True* unless interactive mode is enabled or `%matplotlib`_ is used in IPython.

Reply:
Wouldn't want to go for the long version here. It's just redundant and doesn't add to the clarity.